src/switch_log.c

```diff
@@ -663,6 +666,19 @@ SWITCH_DECLARE(void) switch_log_vprintf(switch_text_channel_t channel, const cha
 		node->slevel = special_level;
 		node->content = content;
 		node->timestamp = now;
+
+		if (!clock_gettime(CLOCK_REALTIME, &tv)) {
+			node->timestamp_nano = tv.tv_sec * 1000000000 + tv.tv_nsec;
+		} else {
+			fprintf(stderr, "Error clock_gettime\n");
+			node->timestamp_nano = 0;
+		}
+
+		switch_mutex_lock(COUNTERLOCK);
```

Comment:
Thanks Ryan, this is true, although that's just a simple incrementation of integer, I think there is no need to make this switchable by modules. If no module uses it it will not cause any damage.

Reply:
My worry is the lock, we have introduce locking on this function and this is heavily being called throughout the FS codes and not sure what would be the impact of this change or this might be negligible..